include/substrait/common/Io.h

```diff
@@ -0,0 +1,30 @@
+/* SPDX-License-Identifier: Apache-2.0 */
+
+#pragma once
+
+#include <string_view>
+
+#include "absl/status/statusor.h"
+#include "substrait/proto/plan.pb.h"
+
+namespace io::substrait {
+
+enum PlanFileEncoding {
+  kBinary = 0,
+  kJson = 1,
+  kProtoText = 2,
+  kText = 3,
+};
+
+// Loads a Substrait plan consisting of any encoding type from the given file.
+absl::StatusOr<::substrait::proto::Plan> loadPlanWithUnknownEncoding(
```

Comment:
This pushes abseil all the way out to the public API?  Could we keep it an internal private dependency?  Maybe by doing something clumsier like...

```
std::optional<std::string> loadPlanWithUnknownEncoding(std::string_view input_filename, ::substrait::proto::Plan* out);
```

Reply:
For now I'll leave this as is, I'll ask a question at the community meeting whether we'd prefer this or have something custom until C++23's std::expected implementation arrives.